### DISCUSSION TITLE
HW modification for jitter-free calibration frequency out

### DISCUSSION THREAD

**Discussion by Ho-Ro** [OpenHantek/OpenHantek6022#283] [Show and tell]
The calibration output is generated by toggling an output pin in an interrupt routine triggered by counter/timer 2 overflow. Due to interrupt latency the signal jitters by some µs, this leads to a high phase jitter on high output frequencies. The EzUSB can provide the T2 overflow signal on pin PE2/T2OUT as an alternate port function (PE2 is available on JP2 of the PCB, at least for 6022BE, for 6022BL you must solder directly on pin 88 of the 100-pin TQFP EzUSB chip). According to TRM (rev. G, p.193) the signal is *"one CLKOUT pulse"*. Add a D-flipflop (e.g. 74HC74) to create the correct frequency with a 50% duty cycle. 
![image](https://user-images.githubusercontent.com/12542359/161778175-c368556e-baa9-4f5e-bed0-04bf35e97a4c.png)
Depending on your needs, supply the flipflop either with 5V (pin 3 of U5 - AMS1117-3.3) or with 3.3V (pin 2 of U5), GND is U5/pin1.
More info in the FW discussion [#16](https://github.com/Ho-Ro/Hantek6022API/discussions/16).
This FW mod works perfectly with my 6022BE and is available in Openhantek for 6022BE and BL since commit [5973a32](https://github.com/OpenHantek/OpenHantek6022/commit/5973a32a0201af8caf9262f3f26c77a6d73b0d1c).

**Comment by cryptozoy** [comment]
Binary counters like 74HC161 or 74HC163 can be used here too. Install what you have. ;-)

**Reply by Ho-Ro** [reply]
> Install what you have. ;-)

I did, in my scope works a 74HC4024 (7-bit ripple counter, Q0 used), the 1st DIL counter I found in my junk box.
